lib/chef/knife/knife_ec_error_handler.rb

```diff
@@ -0,0 +1,66 @@
+# Author:: Jeremy Miller (<jmiller@chef.io>)
+# Copyright:: Copyright (c) 2017 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class Chef
+  class Knife
+    # EcErrorHandler
+    class EcErrorHandler
+      require 'tempfile'
+
+      def initialize
+        @errors_file = Tempfile.new
```

Comment:
Would it make more sense to have the error file inside the directory we are backing up or restoring? That way we could have it persistent for further troubleshooting or even just to go through the failures in the future and not just printing them out. 

This correlates with the idea that @marcparadise had about creating specific error files per object, though that might come on a next iteration but we should start with a persistent file at least.

Maybe: 

- Backup: `"#{dest_dir}/errors/backup-#{date}.json`
- Restore: `"#{dest_dir}/errors/restore-#{date}.json`

Thoughts?